src/app/attendance/components/AttendanceFormBuilder.tsx

```diff
@@ -0,0 +1,263 @@
+    shift: "C",
+    role: "shift-incharge",
+    status: "active",
+  },
+  { id: 4, name: "Diana Prince", shift: "D", role: "member", status: "active" },
+  {
+    id: 5,
+    name: "Ethan Hunt",
+    shift: "A",
+    role: "incharge",
+    status: "invited",
+  },
+  {
+    id: 6,
+    name: "Fiona Gallagher",
+    shift: "B",
+    role: "member",
+    status: "active",
+  },
+  {
+    id: 7,
+    name: "George Banks",
+    shift: "B",
+    role: "shift-incharge",
+    status: "active",
+  },
+  {
+    id: 8,
+    name: "Hannah Baker",
+    shift: "B",
+    role: "member",
+    status: "invited",
+  },
+  {
+    id: 9,
+    name: "Ian Malcolm",
+    shift: "C",
+    role: "incharge",
+    status: "active",
+  },
+  {
+    id: 10,
+    name: "Jack Sparrow",
+    shift: "C",
+    role: "member",
+    status: "active",
+  },
+  {
+    id: 11,
+    name: "Kara Danvers",
+    shift: "D",
+    role: "shift-incharge",
+    status: "invited",
+  },
+];
+
+const AttendanceFormBuilder = () => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    setValue,
+    getValues,
+  } = useForm<AttendanceFormValues>({
+    resolver: zodResolver(attendanceSchema),
+    defaultValues: {
+      memberId: loginUser.role === "member" ? loginUser.id : undefined,
+    },
+  });
+
+  const handleClick = (event: React.MouseEvent<HTMLInputElement>) => {
+    event.currentTarget.showPicker(); 
+  };
+
+  const roleBaseMembers = useMemo(() => {
+    return members
+      .filter(
+        (member) =>
+          member.status === "active" &&
+          (loginUser.role === "incharge" || member.shift === loginUser.shift)
+      )
+      .map(({ id, name }) => ({
+        id,
+        name,
+      }));
+  }, []);
+
+  const onSubmit = (values: AttendanceFormValues) => {
+    console.log({values});
+    const payload = {
+      ...values,
+      // status: "pending", // we will add this status pending in the sever action 
+    }
+  };
+
+  const handleUserSelect = (value: string) => {
+    const userId = Number(value);
+    setValue("memberId", userId, { shouldValidate: true });
+  };
+
+  return (
+    <div className="max-w-lg mx-auto p-8 mt-10 bg-white shadow-md rounded-md">
+      <h1 className="text-2xl font-bold text-center mb-6">Attendance Form</h1>
+
+      <form onSubmit={handleSubmit(onSubmit)} className="flex flex-col gap-6">
+        <div className="flex flex-col">
+          <label htmlFor="user" className="text-sm font-medium">
+            User name
+          </label>
+          <Select onValueChange={handleUserSelect}>
+            <SelectTrigger
+              className={`mt-1 border rounded-md p-2 ${
+                errors.memberId ? "border-red-500" : "border-gray-300"
+              }`}
+              disabled={loginUser.role === "member"}
+            >
+              <SelectValue
+                placeholder={
+                  loginUser.role === "member" ? loginUser.name : "select a user"
+                }
```

Comment:
@ghulamhafeez  I think there no need `loginUser.name` placeholder  because it value exist.